pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -760,6 +802,83 @@ func (pl *DynamicResources) PreBind(ctx context.Context, cs *framework.CycleStat
 			state.claims[index] = claim
 		}
 	}
+
+	if !pl.enableDeviceBindingConditions {
+		// If we get here, we know that reserving the claim for
+		// the pod worked and we can proceed with binding it.
+		return nil
+	}
+
+	// We need to check if the device is attached to the node.
+	needWait := false
+	for _, claim := range state.claims {
+		for _, device := range claim.Status.Devices {
+			if device.BindingConditions != nil {
+				needWait = true
+				break
+			}
+		}
+	}
+
+	// If no device needs to be prepared, we can return early.
+	if !needWait {
+		return nil
+	}
+
+	// We need to decide how long we should wait for the device to be attached to the node.
+	timeout := 10 * time.Minute
+	for _, claim := range state.claims {
+		for _, device := range claim.Status.Devices {
+			if device.BindingTimeout != nil && timeout < device.BindingTimeout.Duration {
+				timeout = device.BindingTimeout.Duration
+			}
+		}
+	}
+
+	// We need to wait for the device to be attached to the node.
```

Comment:
@pohly does this pause all scheduling or is this run in a go routine somewhere? We shouldn't be pausing all scheduling to wait for this.